Sources/SwiftProtobuf/ProtobufJSONTypes.swift

```diff
@@ -818,12 +818,7 @@ public extension ProtobufJSONMessageBase {
     }
 
     func serializeAnyJSON() throws -> String {
-        var jsonVisitor = ProtobufJSONEncodingVisitor()
-        try jsonVisitor.withAbstractVisitor {(visitor: inout ProtobufVisitor) in
-            try visitor.visitSingularField(fieldType: ProtobufString.self, value: anyTypeURL, protoFieldNumber: 1, protoFieldName: "type_url", jsonFieldName: "@type", swiftFieldName: "typeURL")
-            try traverse(visitor: &visitor)
-        }
-        return jsonVisitor.result
+        return try ProtobufJSONEncodingVisitor(message: self, anyTypeURL: anyTypeURL).result
```

Comment:
As above, I've started thinking of `AnyJSON` as a distinct serialization.  The original code here leveraged the JSON encoder to support most of the fields, but didn't ask the regular JSON encoder to handle the special `@type` member.